packages/vite/src/node/optimizer/scan.ts

```diff
@@ -300,25 +297,9 @@ function esbuildScanPlugin(
                 (loader.startsWith('ts') ? extractImportPaths(content) : '')
 
               const key = `${path}?id=${scriptId++}`
-
-              if (contents.includes('import.meta.glob')) {
-                scripts[key] = {
-                  // transformGlob already transforms to js
-                  loader: 'js',
-                  contents: await transformGlob(
-                    contents,
-                    path,
-                    config.root,
-                    loader,
-                    resolve,
-                    config.logger
-                  )
-                }
-              } else {
-                scripts[key] = {
-                  loader,
-                  contents
-                }
+              scripts[key] = {
+                loader,
+                contents
```

Comment:
This is the one place that I'm still unsure it won't affect us. It is strange to me that if a user has:
```js
import 'module-a.js'
import 'module-b.js'
```
And modifies the code to
```js
import.meta.glob('module-?.js', { eager: true })`
```
Then scanning would be affected.

Or is this already taken care of in the PR? 🤔
Would it be possible to run the glob plugin when `import.plugin.glob` is present before this analysis if not?